component/src/main/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestAuthenticator.java

```diff
@@ -285,12 +291,15 @@ private JSONObject fetchUserInfo(String url, String accessToken) throws Applicat
 		BufferedReader bufferedReader = null;
 		HttpURLConnection httpConnection = null;
 		JSONObject jsonObj = null;
+
 		try {
-			URL obj = new URL(url + "?" + PinterestAuthenticatorConstants.ACCESS_TOKEN + "=" + accessToken);
+			URL obj = new URL(url);
 			URLConnection connection = obj.openConnection();
+
 			// Cast to a HttpURLConnection
 			if (connection instanceof HttpURLConnection) {
 				httpConnection = (HttpURLConnection) connection;
+				httpConnection.setRequestProperty("Authorization", "Bearer " + accessToken);
```

Comment:
User constants for "Authorization" and "Bearer"

Reply:
Fixes: https://github.com/wso2-extensions/identity-outbound-auth-pinterest/pull/14